OSSM-5690: Sync with istio 1.20

yannuil opened the pull request.

yannuil committed:
Update deps

Signed-off-by: Yann Liu <yannliu@redhat.com>

yannuil committed:
Update API groups

Signed-off-by: Yann Liu <yannliu@redhat.com>

yannuil committed:
Run make gen

Signed-off-by: Yann Liu <yannliu@redhat.com>

yannuil committed:
Use the same version gateway-api as in upstream 1.20

Signed-off-by: Yann Liu <yannliu@redhat.com>

yannuil committed:
Update openshift packages

Signed-off-by: Yann Liu <yannliu@redhat.com>

yannuil committed:
Use the same istio client-go as in the upstream 1.20

Signed-off-by: Yann Liu <yannliu@redhat.com>

jewertow commented on go.mod:
```diff
@@ -1,72 +1,75 @@
-	k8s.io/gengo v0.0.0-20220902162205-c0856e24416d
-	k8s.io/klog/v2 v2.90.1
-	sigs.k8s.io/gateway-api v0.6.2
+	istio.io/client-go v1.20.0-rc.0
```
Shouldn't it be `istio.io/client-go v1.20.2-0.20231213021015-d9f0411d8b67`, like in [maistra/istio](https://github.com/maistra/istio/pull/920/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R113)?

yannuil replied:
I want to make it the same as the istio [commit](https://github.com/istio/istio/commit/6869a6c2371e21c847d216065cf5c59863d01b4c) 1.20 I am rebasing on, I can eliminate the extra changes in the go.mod commit.

yannuil committed:
Update depguard configuration as a golangci limitation

For details, please check out this [PR](https://github.com/golangci/golangci-lint/pull/4227).

Signed-off-by: Yann Liu <yannliu@redhat.com>

yannuil commented:
/retest-required

yannuil commented:
/retest-required

jwendell commented:
/retest

jwendell commented:
/retest
